pkg/asset/manifests/storageclassconfig_test.go

```diff
@@ -0,0 +1,49 @@
+package manifests
+
+import (
+	"testing"
+
+	"github.com/ghodss/yaml"
+	"github.com/stretchr/testify/assert"
+
+	storagev1 "k8s.io/api/storage/v1"
+
+	azuretypes "github.com/openshift/installer/pkg/types/azure"
+)
+
+// TestStorageClassConfig tests that a valid storage class manifest is created.
+func TestStorageClassConfig(t *testing.T) {
+	createStorageClassConfig := func() *storagev1.StorageClass {
+		machinePool := &azuretypes.MachinePool{
+			OSDisk: azuretypes.OSDisk{
+				DiskType: "Premium_LRS",
+				DiskEncryptionSet: &azuretypes.DiskEncryptionSet{
+					SubscriptionID: "08675309-1111-2222-3333-303606808909",
+					ResourceGroup:  "test-resource-group",
+					Name:           "test-encryption-set",
+				},
+			},
+		}
+		return validAzureStorageClass(machinePool, "master")
+	}
+
+	expectedConfig := createStorageClassConfig()
+	expectedYaml := `allowVolumeExpansion: true
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  creationTimestamp: null
+  name: master-managed-premium-encrypted-cmk
+parameters:
+  diskencryptionsetid: /subscriptions/08675309-1111-2222-3333-303606808909/resourceGroups/test-resource-group/providers/Microsoft.Compute/diskEncryptionSets/test-encryption-set
+  kind: Managed
+  resourcegroup: test-resource-group
+  skuname: Premium_LRS
+provisioner: kubernetes.io/azure-disk
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer
+`
```

Comment:
This test concerns me a little because it seems like we would need to change the test to accommodate changes to the code. This is probably manageable if we don't expect the code to change much.

Reply:
I don't see this being a problem.